### PR TITLE
Add WorldEdit version and platform to schematics

### PR DESCRIPTION
**Pull request by me4502**
Add the WorldEdit version and metadata into the metadata of Sponge schematics. Could be useful for debugging in the future.

### CODE REVIEW

File: worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlatform.java [hunk 1]
```diff
@@ -84,4 +85,8 @@ public void reload() {
         getTranslationManager().reload();
     }
 
+    @Override
+    public String getId() {
+        return "legacy:" + getPlatformName().toLowerCase(Locale.ROOT).replace(" ", "_");
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlatform.java:90]: need to sub everything outside `[a-z_.-]` with `_` probably, not just spaces, for full transformation.

File: worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/SpongeSchematicWriter.java [hunk 2]
```diff
@@ -114,6 +115,23 @@ public void write(Clipboard clipboard) throws IOException {
...
         metadata.put("WEOffsetZ", new IntTag(offset.getBlockZ()));
 
+        Map<String, Tag> worldEditSection = new HashMap<>();
+        worldEditSection.put("Version", new StringTag(WorldEdit.getVersion()));
+        worldEditSection.put("EditingPlatform", new StringTag(WorldEdit.getInstance().getPlatformManager().queryCapability(Capability.WORLD_EDITING).getId()));
+        worldEditSection.put("Offset", new IntArrayTag(new int[]{offset.getBlockX(), offset.getBlockY(), offset.getBlockZ()}));
+
+        Map<String, Tag> platformsSection = new HashMap<>();
+        for (Platform platform : WorldEdit.getInstance().getPlatformManager().getPlatforms()) {
+            Map<String, Tag> platformSection = new HashMap<>();
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/SpongeSchematicWriter.java:125]: Consider `ImmutableMap` or `CompoundTagBuilder` here.

**Review by octylFractal — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlatform.java [hunk 3]
```diff
@@ -32,6 +32,7 @@
 import java.io.UncheckedIOException;
 import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlatform.java:35]: make diff clean plz

File: worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/Platform.java [hunk 4]
```diff
@@ -178,6 +181,12 @@
      */
     String getPlatformVersion();
 
+    @NonAbstractForCompatibility(delegateName = "getPlatformName", delegateParams = {})
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/Platform.java:184]: Add the `@apiNote` from the other like-annotated methods too please

**Review by octylFractal — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/Platform.java [hunk 5]
```diff
@@ -181,6 +181,12 @@
      */
     String getPlatformVersion();
 
+    /**
+     * The id of this object in the registry. Must be unique, and lowercase. Certain registries (e.g Namespaced ones) may have additional restrictions.
```
Comment (octylFractal) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/Platform.java:185]: have you heard of `{@inheritDoc}`

**Review by octylFractal — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/Platform.java (hunk 5 above)
Comment (me4502) [reply worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/Platform.java:185]: s m h

**Review by octylFractal — APPROVED** [review]

**Pull request merged**